_drafts/oryx/dependency-injection/oryx-service-layer.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Oryx service layer
+description: Oryx service layer abstracts system functionality using DI
+template: concept-topic-template
+last_updated: Apr 13, 2023
+---
+
+The service layer in Oryx serves as the foundation for the business logic, with Dependency injection(DI) playing a crucial role.
+
+The main objective of the service layer in Oryx is to abstract all the system functionality, including querying backend services, data caching and reloading, state management, and reactivity.
+
+Furthermore, DI is utilized by many higher-level elements and concepts in Oryx, ranging from adapters and normalizers to HTTP interceptors, authentication, product loading, cart management, and checkout process.
+
+## Dependency injection npm package
+
+The `@spryker-oryx/di` package provides DI not only to Oryx applications but also to other packages and can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with DI, such as the following:
```

Comment:
Optional suggestion:
```suggestion
The `@spryker-oryx/di` package provides DI to Oryx applications and other packages. It can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with DI—for example:
```